Improve Document Viewer error handling and state management

FlyJoanne opened the pull request:
- Add user-friendly messages when no PDF files are available
- Fix state management to properly clear content when switching entries
- Replace technical exceptions with helpful placeholder text
- Improve user experience for entries with non-PDF files only

Closes #13198

Improved the Document Viewer's handling of entries that contain only non-PDF files by replacing technical error messages with user-friendly placeholders and fixing state management issues that caused stale content to be displayed when switching between entries.

### Steps to test

1. **Test entries with non-PDF files only:**
   - Select a bibliography entry that has only .docx, .md, or other non-PDF files attached
   - Open Document Viewer
   - Expected: Should display "No PDF available for preview" message instead of technical exceptions

![image](https://github.com/user-attachments/assets/cddcddaa-b3b1-44ee-be72-97c1a87b04bf)

2. **Test normal PDF functionality:**
   - Select an entry with PDF files attached
   - Open Document Viewer
   - Expected: PDF should display normally with dropdown showing only PDF files

<img width="803" alt="image" src="https://github.com/user-attachments/assets/d04ecb90-718f-404e-8bde-3a1ede345d12" />

3. **Test state management:**
   - First open Document Viewer with an entry containing PDF files
   - Then switch to an entry with only non-PDF files
   - Expected: Should show placeholder message, not the previous PDF content

![image](https://github.com/user-attachments/assets/b1db8d70-4e76-474f-af76-5d1b1ece206e)

### Mandatory checks

- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if change is visible to the user)
- [/] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (if change is visible to the user)
- [/] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [/] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

Siedlerchr commented on jabgui/src/main/java/org/jabref/gui/documentviewer/DocumentViewerViewModel.java:
```diff
@@ -76,10 +80,29 @@ public StringProperty highlightTextProperty() {
+
+            // Filter to include only PDF files
+            Set<LinkedFile> pdfFiles = linkedFiles.stream()
+                                                  .filter(this::isPdfFile)
```
Move idPdfFile filter in the stream above so you can directlry filter

Siedlerchr commented on jabgui/src/main/java/org/jabref/gui/documentviewer/DocumentViewerViewModel.java:
```diff
@@ -76,10 +80,25 @@ public StringProperty highlightTextProperty() {
+            if (pdfFiles.isEmpty()) {
+                // No PDF files found - clear the list and current document
+                files.clear();
+                currentDocument.set(null);
```
I htink it would be good display a notification: dialogService.notify(Localization.lang("No pdf files  available"))

Siedlerchr commented:
Just one little addition and I think it's ready

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/DocumentViewerViewModel.java:
```diff
@@ -89,11 +108,34 @@ private void setCurrentDocument(Path path) {
+        try {
+            Path filePath = Path.of(file.getLink());
+            return FileUtil.isPDFFile(filePath);
+        } catch (Exception e) {
```
do not catch generic exceptions, be very specific. You don't know then why sthg failed? 1000% sure it is because it is not a pdf file?

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/PdfDocumentViewer.java:
```diff
@@ -25,13 +27,27 @@ public class PdfDocumentViewer extends StackPane {
+
+        // Create placeholder label
+        placeholderLabel = new Label("No PDF available for preview");
+        placeholderLabel.setStyle("-fx-text-fill: #666666; -fx-font-size: 16px;");
```
no random style definitions in code. Put everything in base.css !

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/PdfDocumentViewer.java:
```diff
@@ -25,13 +27,27 @@ public class PdfDocumentViewer extends StackPane {
-        getChildren().add(pdfView);
+
+        // Create placeholder label
+        placeholderLabel = new Label("No PDF available for preview");
```
Localization

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/PdfDocumentViewer.java:
```diff
@@ -25,13 +27,27 @@ public class PdfDocumentViewer extends StackPane {
+
+        getChildren().addAll(pdfView, placeholderLabel);
+
+        // Initially show placeholder
+        pdfView.setVisible(false);
+        placeholderLabel.setVisible(true);
+
         EasyBind.subscribe(currentPage, current -> pdfView.setPage(current.intValue()));
         // We can only set the search query at the moment not the results or mark them in the text
         EasyBind.subscribe(highlightText, pdfView::setSearchText);
+        // Initially hide PDFView until a document is loaded
+        pdfView.setVisible(false);
```
duplication

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/PdfDocumentViewer.java:
```diff
@@ -47,11 +63,23 @@ public void show(Path document) {
             try {
                 pdfView.load(Files.newInputStream(document));
                 pdfView.setPage(currentPage.get());
+                // Show PDF and hide placeholder
+                pdfView.setVisible(true);
+                placeholderLabel.setVisible(false);
```
meaningless comment. superfluous. But i know. AI loves this stupid comments.
What LLM model do you use?

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/PdfDocumentViewer.java:
```diff
@@ -47,11 +63,23 @@ public void show(Path document) {
+                LOGGER.debug("Successfully loaded PDF document: {}", document);
             } catch (IOException e) {
                 LOGGER.error("Could not load PDF document {}", document, e);
+                // Show error message
+                pdfView.setVisible(false);
+                placeholderLabel.setText("Could not load PDF: " + document.getFileName());
+                placeholderLabel.setVisible(true);
```
Localization

calixtus commented on jabgui/src/main/java/org/jabref/gui/documentviewer/DocumentViewerView.java:
```diff
@@ -99,9 +99,8 @@ private void setupFileChoice() {
-            if (newDocument != null) {
-                viewer.show(newDocument);
-            }
+            // Always call show(), even when newDocument is null
+            viewer.show(newDocument);
```
meaningless comment

calixtus commented:
https://devdocs.jabref.org/code-howtos/faq.html#submodules

https://github.com/JabRef/jabref/blob/main/CONTRIBUTING.md#notes-on-ai-usage

FlyJoanne commented:
@calixtus @Siedlerchr Thank you for all the detailed feedback throughout this PR. I've put considerable effort into addressing the review comments and just pushed one final fix for the state management issue.

The Document Viewer now properly handles all scenarios in manual testing

At this point, I feel I've addressed the main issues to the best of my current ability and would like to step back from this PR.

Feel free to:
- Close this PR if any remaining issues are blockers
- Have a maintainer take over any final adjustments
- Let another contributor continue from here

Thank you for the excellent learning opportunity and all the time you've invested in reviewing! 🙏

Siedlerchr commented:
@FlyJoanne Sorry to hear that, you were nearly finished, just a few minor things. Please don't take comments personally. We encourage newcomers, always, to try to understand the codebase and to take it as an opportunity to learn about the software development process and try to be encouraging, however, in the end, we are also just humans and not perfect ;)
